src/common/transformations/src/transformations/common_optimizations/loop_inputs.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/loop_inputs.hpp"
+
+#include <unordered_map>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/core/type.hpp"
+#include "openvino/op/loop.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace std;
+using namespace ov::element;
+using namespace ov::pass::pattern;
+
+namespace {
+std::shared_ptr<ov::op::v0::Parameter> get_parent_param(const std::shared_ptr<ov::op::v0::Result>& result) {
+    return ov::as_type_ptr<ov::op::v0::Parameter>(result->input_values()[0].get_node_shared_ptr());
+}
+}  // namespace
+
+ov::pass::LoopInputs::LoopInputs() {
+    MATCHER_SCOPE(LoopInputs);
+
+    auto loop_label = wrap_type<ov::op::v5::Loop>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        auto loop = as_type_ptr<ov::op::v5::Loop>(pattern_to_output.at(loop_label).get_node_shared_ptr());
+        auto body_model = loop->get_function();
+        const auto body_params = body_model->get_parameters();
+        const auto body_results = body_model->get_results();
+
+        const auto loop_input_values = loop->input_values();
+        const auto& trip_count = loop_input_values[0];
+        const auto& exec_cond = loop_input_values[1];
+
+        auto new_loop = make_shared<ov::op::v5::Loop>(trip_count, exec_cond);
+        new_loop->set_function(body_model);
+        new_loop->set_special_body_ports(loop->get_special_body_ports());
+
+        for (const auto& input_description : loop->get_input_descriptions()) {
+            if (const auto merged_input_desc =
+                    dynamic_pointer_cast<ov::op::util::MultiSubGraphOp::MergedInputDescription>(input_description)) {
```

Comment:
mixing as_type_ptr (line 35)  and dynamic_pointer_cast (line 50)
please select one method